### PR TITLE
revert vscode dependency

### PR DESCRIPTION
**Pull request by lbenie**
- [x] Code is up-to-date with the `master` branch

Fix vscode dependency version.

Fix some logic with the builders

Fix command name prefix

Add tsling configuration

### CODE REVIEW

File: src/commands/soqlCommand.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import vscode = require('vscode');
 import {ICommand}  from './command';
 import {Connection, IQueryResult} from '../connection';
+// import * as fs from 'fs';
```
Comment (tedre191) [inline src/commands/soqlCommand.ts]: Remove if not used

File: src/commands/soqlCommand.ts [hunk 2]
```diff
@@ -15,13 +16,17 @@ export class SOQLCommand implements ICommand {
         this.conn.executeQuery(query)
           .then((results: IQueryResult) => {
             let json = JSON.stringify(results, null, 2);
+
+            // vscode.commands.executeCommand('vscode.previewHtml', encodeURI(json));
```
Comment (tedre191) [inline src/commands/soqlCommand.ts]: same

File: src/utils.ts [hunk 3]
```diff
@@ -119,6 +119,15 @@ export function getNamespaceFromConfig(): string {
 }
 
 /**
+ * Get the organization from the configuration settings
+ *
+ * @return {string} organization
+ */
+export function getOrganizationFromConfig(): string {
+  return configuration.get<string>('organization');
```
Comment (tedre191) [inline src/utils.ts]: What's in the "organization" field ?

File: src/builder/soqlQueryBuilder.ts [hunk 4]
```diff
@@ -20,13 +20,9 @@ export class SOQLQueryBuilder implements IBuilder {
...
-  private buildSelect(attributes: string[]): SOQLQueryBuilder {
-    attributes.forEach((value, index) => {
-      if (index > 0) {
-        this.query += `, ${value}`;
-      } else {
-        this.query += ` ${value}`;
-      }
+  private buildSelect(attributes): SOQLQueryBuilder {
+    attributes.forEach((value: string, index: number) => {
+      this.query += `${index > 0 ? ',' : ''} ${value}`;
```
Comment (erocheleau) [inline src/builder/soqlQueryBuilder.ts]: :sooneliner:
```
if (index > 0) this.query += ',';
this.query += ` ${value}`;
```

File: src/builder/soqlQueryBuilder.ts [hunk 5]
```diff
@@ -39,13 +35,9 @@ export class SOQLQueryBuilder implements IBuilder {
...
-  private buildFrom(databases: string[]): SOQLQueryBuilder {
-    databases.forEach((value, index) => {
-      if (index > 0) {
-        this.query += `, ${value}`;
-      } else {
-        this.query += ` FROM ${value}`;
-      }
+  private buildFrom(tables): SOQLQueryBuilder {
+    tables.forEach((value: string, index: number) => {
+      this.query += `${index > 0 ? ',' : ' FROM'} ${value}`;
```
Comment (erocheleau) [inline src/builder/soqlQueryBuilder.ts]: 👍

File: src/commands/soqlCommand.ts (hunk 1 above)
Comment (erocheleau) [inline src/commands/soqlCommand.ts]: 👍

File: src/commands/soqlCommand.ts [hunk 6]
```diff
@@ -15,13 +16,17 @@ export class SOQLCommand implements ICommand {
         this.conn.executeQuery(query)
           .then((results: IQueryResult) => {
```
Comment (erocheleau) [inline src/commands/soqlCommand.ts]: Are these promises or Thenables?

```
 vscode.window.showInputBox({ prompt: 'Query: ' })
 .then((query: string) => {
     return this.conn.executeQuery(query);
}, (reason) => {})
.then((results: IQueryResult) => {...));
```

File: src/commands/soqlCommand.ts [hunk 7]
```diff
@@ -15,13 +16,17 @@ export class SOQLCommand implements ICommand {
...
             let json = JSON.stringify(results, null, 2);
+
+            // vscode.commands.executeCommand('vscode.previewHtml', encodeURI(json));
             let outputConsole = vscode.window.createOutputChannel('SOQL query');
             outputConsole.appendLine('Query Results');
             outputConsole.appendLine(json);
             outputConsole.show();
 
             // TODO: use preview.html or open commands
```
Comment (erocheleau) [inline src/commands/soqlCommand.ts]: TODO?

**Pull request merged**